components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/NewApiVersionMailNotifier.java

```diff
@@ -0,0 +1,150 @@
+ *  under the License.
+ *
+ */
+package org.wso2.carbon.apimgt.core.impl;
+
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.core.configuration.models.APIMConfigurations;
+import org.wso2.carbon.apimgt.core.configuration.models.MailConfigurations;
+import org.wso2.carbon.apimgt.core.exception.APIManagementException;
+import org.wso2.carbon.apimgt.core.template.dto.NotificationDTO;
+
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Properties;
+import java.util.Set;
+import javax.mail.Authenticator;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.PasswordAuthentication;
+import javax.mail.Session;
+import javax.mail.Transport;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+
+
+/**
+ * Class to send Email notification.
+ */
+public class NewApiVersionMailNotifier extends Notifier {
+    private static final Log log = LogFactory.getLog(NewApiVersionMailNotifier.class);
+    static MailConfigurations mailConfigurations = new APIMConfigurations().
+            getNotificationConfigurations().getMailConfigurations();
+    private static final String SMTP_HOST_NAME = mailConfigurations.getSmtpHostname();
+    private static final String SMTP_AUTH_USER = mailConfigurations.getSmtpAuthUser();
+    private static final String SMTP_AUTH_PWD = mailConfigurations.getSmtpAuthPwd();
+
+
+    @Override
+    public void sendNotifications(NotificationDTO notificationDTO) throws
+            APIManagementException {
+
+        Properties props = new Properties();
+        props.put("mail.smtp.host", SMTP_HOST_NAME);
+        props.put("mail.smtp.auth", "true");
+        props.put("mail.smtp.starttls.enable", "true");
+        props.put("mail.smtp.port", "587");
+        props.put("mail.transport.protocol", "smtp");
+        //get Notifier email List
+        Set<String> sendTo = getEmailNotifierList(notificationDTO);
+
+        if (sendTo.isEmpty()) {
+            log.debug("Email Notifier Set is Empty");
+            return;
+        }
+        for (String mail : sendTo) {
+            try {
+                Authenticator auth = new SMTPAuthenticator();
+                Session mailSession = Session.getDefaultInstance(props, auth);
+                MimeMessage message = new MimeMessage(mailSession);
+                notificationDTO.setTitle((String) notificationDTO.getProperty(NotifierConstants.TITLE_KEY));
+                notificationDTO.setMessage((String) notificationDTO.getProperty(NotifierConstants.TEMPLATE_KEY));
+                notificationDTO = loadMailTemplate(notificationDTO);
+                message.setSubject(notificationDTO.getTitle());
+                message.setContent(notificationDTO.getMessage(), NotifierConstants.TEXT_TYPE);
+                message.setFrom(new InternetAddress(mailConfigurations.getFromUser()));
+
```

Comment:
i think we don't need spaces here

Reply:
Fixed with PR: https://github.com/wso2/carbon-apimgt/pull/4537